Lib/distutils/sysconfig.py

```diff
@@ -193,7 +193,7 @@ def customize_compiler(compiler):
 
         if 'CC' in os.environ:
             newcc = os.environ['CC']
-            if (sys.platform == 'darwin'
+            if (os.name == 'posix'
                     and 'LDSHARED' not in os.environ
                     and ldshared.startswith(cc)):
                 # On OS X, if CC is overridden, use that as the default
```

Comment:
If you change the test to include all posix platforms, this comment needs to be changed as well.

Reply:
Nice catch! Updated!